src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
 
+    public void push(T value) {
+        Node<T> node;
+        if (size == 0) {
+            node = new Node<>(value, null);
+        } else {
+            node = new Node<>(value, top);
+        }
+        top = node;
```

Comment:
```suggestion
        top = new Node<>(value, top);
```

Reply:
done

Reply:
I've removed block if - else here, because is size ==0, top ==null